3rdparty/mshadow/CMakeLists.txt

```diff
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 2.8.7)
+cmake_minimum_required(VERSION 3.0.2)
```

Comment:
For cuda nothing less than 3.12 seems to work in ubuntu 18.04

Reply:
This file only adds the lint target. Let's just keep it unchanged for now thanks to 

> CMake keeps policy settings on a stack, so changes made by the cmake_policy command affect only the top of the stack. A new entry on the policy stack is managed automatically for each subdirectory to protect its parents and siblings.